adal/src/androidTest/java/com/microsoft/aad/adal/AcquireTokenSilentHandlerTest.java

```diff
@@ -82,6 +88,68 @@ protected void setUp() throws Exception {
     }
 
     /**
+     * Acquire token users refresh token, but the client app is inactive.
+     */
+    @TargetApi(Build.VERSION_CODES.M)
+    public void testRefreshTokenFailedNoNetworkAppIsInactive() {
+        FileMockContext mockContext = new FileMockContext(getContext());
+        mockContext.setAppInactive();
+        ITokenCacheStore mockCache = getCacheForRefreshToken(TEST_IDTOKEN_USERID, TEST_IDTOKEN_UPN);
+
+        final String resource = "resource";
+        final String clientId = "clientId";
+        final AuthenticationRequest authenticationRequest = getAuthenticationRequest(VALID_AUTHORITY, resource, clientId, false);
+        authenticationRequest.setUserIdentifierType(UserIdentifierType.UniqueId);
+        authenticationRequest.setUserId(TEST_IDTOKEN_USERID);
+        final AcquireTokenSilentHandler acquireTokenSilentHandler = getAcquireTokenHandler(mockContext,
+                authenticationRequest, mockCache);
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            UsageStatsManagerWrapper mockUsageStatsManagerWrapper = mock(UsageStatsManagerWrapper.class);
+            Mockito.when(mockUsageStatsManagerWrapper.isAppInactive(any(Context.class))).thenReturn(true);
+            UsageStatsManagerWrapper.sInstance = mockUsageStatsManagerWrapper;
+            PowerManagerWrapper mockPowerManagerWrapper = mock(PowerManagerWrapper.class);
+            Mockito.when(mockPowerManagerWrapper.isDeviceIdleMode(any(Context.class))).thenReturn(false);
+            PowerManagerWrapper.sInstance = mockPowerManagerWrapper;
+        }
+
+        try {
+            acquireTokenSilentHandler.acquireTokenWithRefreshToken("refreshToken");
+            fail("Expect exception");
+        } catch (final Exception exception) {
+            assertTrue(exception instanceof AuthenticationException);
+            assertTrue(((AuthenticationException) exception).getCode() == ADALError.NO_NETWORK_CONNECTION_POWER_OPTIMIZATION);
+        }
+    }
+
+    /**
+     * Acquire token users refresh token, but the client app is inactive.
+     */
+    @Test
```

Comment:
Is this the proper annotation to use? (I honestly don't know -- there seems to be a mix of `@SmallTest`, `@Test`, and unannotated test methods: not sure which ought to be used and when 🤔